common/proof_test.go

```diff
@@ -2,14 +2,183 @@ package common_test
 
```

Comment:
Same comment for the file, I think we can use `github.com/stretchr/testify/require` otherwise the tests look good

Reply:
sure